### PR TITLE
Node: allow configuring IPFIX performance parameters

### PR DESCRIPTION
**Pull request by mariomac**
This commit adds a new `[ipfix]` configuration section that allows tuning some
performance parameters for the IPFIX Flow Monitoring. Concretely:
`cache_active_timeout`, `cache_max_flows` and `sampling`. See Open_vSwitch
database schema docs [1] for more details.

You might observ that now, ovs-vsctl is explicitly invoked with `sampling=400`
by default. This does not change the default behavior of the previous
invocation, as `400` is the default value for the `sampling` configuration
value when it is not specified [1].

The changes have been tested with two extra test cases in the `node_test.go`
file, to check that the ovs-vsctl is properly invoked on several cases. Also,
the proper configuration parsing and overriding is tested in the
`config_test.go` file.

Related JIRA issue: https://issues.redhat.com/browse/NETOBSERV-67

[1] https://www.openvswitch.org/support/dist-docs/ovs-vswitchd.conf.db.5.html

Signed-off-by: Mario Macias <mmaciasl@redhat.com>

### CODE REVIEW

File: contrib/kind.sh [hunk 1]
```diff
@@ -45,41 +47,44 @@ usage() {
...
+    echo "                                    DEFAULT: Remove taint components."
+    echo "-ha  | --ha-enabled                 Enable high availability. DEFAULT: HA Disabled."
+    echo "-ho  | --hybrid-enabled             Enable hybrid overlay. DEFAULT: Disabled."
+    echo "-ds  | --disable-snat-multiple-gws  Disable SNAT for multiple gws. DEFAULT: Disabled."
+    echo "-dp  | --disable-pkt-mtu-check      Disable checking packet size greater than MTU. Default: Disabled"
+    echo "-nf  | --netflow-targets            Comma delimited list of ip:port netflow collectors. DEFAULT: Disabled."
+    echo "-sf  | --sflow-targets              Comma delimited list of ip:port sflow collectors. DEFAULT: Disabled."
+    echo "-if  | --ipfix-targets              Comma delimited list of ip:port ipfix collectors. DEFAULT: Disabled."
+    echo "-ifs | --ipfix-sampling             Rate at which packets should be sampled and sent to each target collector. DEFAULT: 400."
+    echo "-ifm | --ipfix-cache-max-flows      Maximum number of IPFIX flow records that can be cached at a time. If 0, caching is disabled. DEFAULT: Disabled."
```
Comment (jotak) [inline contrib/kind.sh:62]: is it the current behaviour, cache being disabled?

File: go-controller/pkg/config/config.go [hunk 2]
```diff
@@ -238,6 +246,23 @@ type MonitoringConfig struct {
...
+// IPFIXConfig holds IPFIX-related performance configuration options. It requires that the ipfix-targets
+// value of the [monitoring] section contains at least one endpoint.
+type IPFIXConfig struct {
+	// Sampling is an optional integer in range 1 to 4,294,967,295. It holds the rate at which
+	// packets should be sampled and sent to each target collector. If not specified, defaults to
+	// 400, which means one out of 400 packets, on average, will be sent to each target collector.
+	Sampling uint `gcfg:"sampling"`
+	// CacheActiveTimeout is an optional integer in range 0 to 4,200. It holds the maximum period in
+	// seconds for which an IPFIX flow record is cached and aggregated before being sent. If not
+	// specified, defaults to 60. If 0, caching is disabled.
```
Comment (jotak) [inline go-controller/pkg/config/config.go:258]: same as above, I'd like to clarify if both `CacheActiveTimeout` and `CacheMaxFlows` needs to be 0 to disable caching.

Comment (mariomac) [reply go-controller/pkg/config/config.go:258]: Added entry for the next sync meeting

**Comment by mariomac** [comment]
In order to ease the manual testing of the whole feature, I will merge this PR on top of the PR https://github.com/ovn-org/ovn-kubernetes/pull/2649 so we can deploy the ovn-kubernetes with all the required functionalities to work with the CNO.

File: contrib/kind.sh (hunk 1 above)
Comment (amorenoz) [reply contrib/kind.sh:62]: A quick look at the OVS code seems to indicate that if any of those parameters is 0, caching is disabled (behavior is not 100% equal but close enough)

**Pull request closed without merging**